src/tiledbsoma_ml/dataset.py

```diff
@@ -111,43 +107,55 @@ class ExperimentDataset(IterableDataset[Batch]):  # type: ignore[misc]
         experimental
     """
 
-    layer_name: str
-    """``X`` layer to read (within an |ExperimentAxisQuery|'s |Measurement|)."""
-    obs_column_names: Sequence[str]
+    # Core data fields
+    x_locator: XLocator = field()
+    """State required to open an ``X`` |SparseNDArray| (and associated ``obs`` |DataFrame|), within an |Experiment|."""
+    query_ids: QueryIDs = field()
+    """``obs``/``var`` coordinates (from an |ExperimentAxisQuery|) to iterate over."""
+    obs_column_names: List[str] = field(factory=lambda: [*DEFAULT_OBS_COLUMN_NAMES])
     """Names of ``obs`` columns to return."""
-    batch_size: int
+
+    # Configuration fields with defaults
+    batch_size: int = field(default=1, validator=gt(0))
     """Number of rows of ``X`` and ``obs`` data to yield in each |Batch|."""
-    io_batch_size: int
+    io_batch_size: int = field(default=DEFAULT_IO_BATCH_SIZE, validator=gt(0))
     """Number of ``obs``/``X`` rows to fetch together, when reading from the provided |ExperimentAxisQuery|."""
-    shuffle: bool
+    shuffle: bool = field(default=True)
     """Whether to shuffle the ``obs`` and ``X`` data being returned."""
-    shuffle_chunk_size: int
-    """Number of contiguous rows shuffled as an atomic unit (before later concatenation/shuffling within IO-batches)."""
-    seed: int
+    shuffle_chunk_size: int = field(default=DEFAULT_SHUFFLE_CHUNK_SIZE, validator=gt(0))
+    r"""Number of contiguous rows shuffled as an atomic unit (before later concatenation and shuffling within |IOBatch|\
+    s)."""
+    seed: Optional[int] = field(default=None)
     """Random seed used for shuffling."""
-    return_sparse_X: bool
+    return_sparse_X: bool = field(default=False)
     r"""When ``True``, return ``X`` data as a |csr_matrix| (by default, return |ndarray|\ s)."""
-    use_eager_fetch: bool
+    use_eager_fetch: bool = field(default=True)
     """Pre-fetch one "IO batch" and one "mini batch"."""
 
-    def __init__(
-        self,
+    # Internal state
+    epoch: int = field(default=0, init=False)
+    rank: int = field(init=False)
+    world_size: int = field(init=False)
+
+    @classmethod
+    def create(
```

Comment:
Super tiny nit:  changing from a normal class init to a factory method is going to trip up some folks.  Suggestion: can you catch this mistake and give them a useful error?

given how `attrs` is structured, I'm not sure it is simple to do, so entirely optional idea.  But I'll note that most PyTorch Dataset subclasses I've seen (e.g., in Pytorch or Torchvision docs) are initialized via `__init__`  -- i.e., don't use a factory method.

Reply:
I like it!